### PR TITLE
Add stack protector support

### LINKED ISSUE #141: Support stack protector
Compiling with `-fstack-protector` (which is default on many Linux distros now) causes a link failure in mlibc.

### PR DESCRIPTION
**Pull request by Geertiebear**
Fixes #141.

### CODE REVIEW

File: options/rtdl/generic/main.cpp [hunk 1]
```diff
@@ -122,6 +123,7 @@ extern "C" void *interpreterMain(uintptr_t *entry_stack) {
 	size_t phdr_entry_size = 0;
 	size_t phdr_count = 0;
 	void *entry_pointer = 0;
+	void *stack_entropy = 0;
```
Comment (avdgrinten) [inline options/rtdl/generic/main.cpp:126]: `nullptr`.

File: options/internal/gcc/stack_protector.cpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#include <stdint.h>
+#include <string.h>
+#include <mlibc/debug.hpp>
+#include <mlibc/stack_protector.hpp>
+#include <mlibc/thread.hpp>
+#include <bits/ensure.h>
+
+uintptr_t __stack_chk_guard = 0;
```
Comment (avdgrinten) [inline options/internal/gcc/stack_protector.cpp:8]: This belong to an external ABI, right? What's its purpose?

**Review by avdgrinten — COMMENTED** [review]
What about threads other than the main thread? How do we initialize the TCB field for these?

File: meson.build [hunk 3]
```diff
@@ -178,12 +179,14 @@ if not no_headers
...
 
 rtdl_sources += [
+	'options/internal/gcc/stack_protector.cpp',
 	'options/internal/gcc/guard-abi.cpp',
 	'options/internal/generic/allocator.cpp',
 	'options/internal/generic/debug.cpp',
 	'options/internal/generic/ensure.cpp',
 	'options/internal/generic/essential.cpp',
 	'options/internal/generic/frigg.cpp',
+	'options/internal/x86_64/thread.cpp',
```
[the remark this answers is not in the document]
Comment (Geertiebear) [reply meson.build:189]: For access to ``get_current_tcb`` from rtdl

File: options/internal/gcc/stack_protector.cpp (hunk 2 above)
Comment (Geertiebear) [reply options/internal/gcc/stack_protector.cpp:8]: Some platforms/toolchains will emit a reference to ``__stack_chk_guard`` for the stack canary, others will get it from ``fs:0x28``.

Comment (avdgrinten) [reply options/internal/gcc/stack_protector.cpp:8]: Alright. How does `__stack_chk_guard` work in multi-threaded programs?

**Review by avdgrinten — APPROVED** [review]
LGTM.

**Pull request merged**